### DISCUSSION TITLE
How to disable htmx include the closest form for non-GET  request?

### DISCUSSION THREAD

**Discussion by zw963** [bigskysoftware/htmx#2568] [Q&A] [answered]
```   
by default, htmx does not include the closest enclosing form for GET requests, but it does include the form for all other types of requests.
   The `hx-include' attribute allows you to select input values that you wish to include in a request via CSS selectors.
```

I want to avoid it for only summit one of input in the form use `hx-put` instead, so, can we disable this behavior?

Thanks.

**Comment by Telroshan** [comment]
Hey, seems there's no way to prevent this behavior, which is currently implemented as follows:
https://github.com/bigskysoftware/htmx/blob/4bc7c9710e99369a8c659f9b865e75866fd7a446/src/htmx.js#L2609-L2612

As a workaround you may move your button out of your form if it's not meant to target the same action/method anyway

**Reply by zw963** [reply]
Should i create a feature request issue for this? 

following is the use case.

I have a html form, with several checkbox input within it,  what i expected is:

1.  When user submit form as usual, it will submit all input fields value in the form as usual use form action.

2. if user only check/uncheck the checkbox, instead, i expect only checkbox value submit use htmx immediately even user don't need click submit button.

Thanks

**Reply by perttisoomann** [reply]
If I understand it correctly, you have a form, and multiple checkboxes to indicate "on" or "off" status for something, and the form doesn't have any other inputs?

In that case, you might be able to remove the form completely and add the action to checkbox on change. Probably could even ignore any returns/replacing elements, because all you need is "let server know the state changed" but your form already shows the correct state, so you don't need to update HTML.

**Reply by zw963** [reply]
> If I understand it correctly, you have a form, and multiple checkboxes to indicate "on" or "off" status for something, and the form doesn't have any other inputs?

Sorry for confusing, except checkboxs, there are many others inputs there, please check following screenshot.

![image](https://github.com/bigskysoftware/htmx/assets/549126/5ba0d96c-9ed8-493a-9325-3185f9889eaf)

In this case, if i click on checkbox, i don't want the backend server update others input too.

Thanks

**Reply by Telroshan** [reply]
I'm not sure I understand the situation here; why make the checkboxes save their state in the DB on click but not the other inputs?
Isn't that a convoluted and confusing UX? If checkboxes are inside a form along other inputs, I feel like as a user I would expect to have either all of them be saved automatically, or all of them be saved only when I click the submit button, but not some doing the first and some others doing the latter

**Reply by zw963** [reply]
Okay, i am fine with current solution. (save all fields in the form when user only click checkbox)

my init thought just want following (2) can be omit in following user's frequently process:

1. click on checkbox
2.  click submit (omit this)
3. click back to the list page.

**Reply by zw963** [reply, marked as answer]
This default behavior is good, but if it is configurable, it is perfect!
